packages/neuron-wallet/tests/block-sync-render/queue.test.ts

```diff
@@ -0,0 +1,244 @@
+import { Subject } from 'rxjs'
+import { Tip } from '@ckb-lumos/indexer'
+import { AddressType } from '../../src/models/keys/address'
+import AddressGenerator from "../../src/models/address-generator"
+import { AddressPrefix } from '../../src/models/keys/address'
+import SystemScriptInfo from '../../src/models/system-script-info'
+import TransactionWithStatus from '../../src/models/chain/transaction-with-status'
+import {Address, AddressVersion} from '../../src/database/address/address-dao'
+import Queue from '../../src/block-sync-renderer/sync/queue'
+import Transaction from '../../src/models/chain/transaction'
+import TxStatus, {TxStatusType} from '../../src/models/chain/tx-status'
+import Input from '../../src/models/chain/input'
+import Output from '../../src/models/chain/output'
+import OutPoint from '../../src/models/chain/out-point'
+import Script, {ScriptHashType} from '../../src/models/chain/script'
+import { flushPromises } from '../test-utils'
+
+const stubbedIndexerConnectorConstructor = jest.fn()
+const stubbedTxAddressFinderConstructor = jest.fn()
+
+const stubbedConnectFn = jest.fn()
+const stubbedGetChainFn = jest.fn()
+const stubbedGenesisBlockHashFn = jest.fn()
+const stubbedGetTransactionFn = jest.fn()
+const stubbedIpcRenderInvokeFn = jest.fn()
+const stubbedAddressesFn = jest.fn()
+const stubbedSaveFetchFn = jest.fn()
+const stubbedUpdateUsedAddressesFn = jest.fn()
+const stubbedNotifyCurrentBlockNumberProcessedFn = jest.fn()
+const stubbedUpdateCacheProcessedFn = jest.fn()
+
+const stubbedTxAddressFinder = jest.fn().mockImplementation(
+  (...args) => {
+    stubbedTxAddressFinderConstructor(...args)
+    return {
+      addresses: stubbedAddressesFn
+    }
+  }
+)
+
+const stubbedRPCServiceConstructor = jest.fn().mockImplementation(
+  () => ({
+    getChain: stubbedGetChainFn,
+    getTransaction: stubbedGetTransactionFn,
+    genesisBlockHash: stubbedGenesisBlockHashFn
+  })
+)
+
+const resetMocks = () => {
```

Comment:
`jest.resetAllMocks()` should help.

Reply:
Thanks for the suggestion.
However, the `jest.resetAllMocks` somehow is not working well with the mocks from `jest.doMock`, which we use to mock import dependencies.